pennylane/qaoa/cost.py

```diff
@@ -625,8 +691,10 @@ def max_weight_cycle(graph, constrained=True):
         can be prepared using :class:`~.BasisState` or simple :class:`~.PauliX` rotations on the
         ``0`` and ``3`` wires.
     """
-    if not isinstance(graph, nx.Graph):
-        raise ValueError(f"Input graph must be a nx.Graph, got {type(graph).__name__}")
+    if not isinstance(graph, (nx.Graph, rx.PyGraph, rx.PyDiGraph)):
+        raise ValueError(
+            f"Input graph must be a nx.Graph or rx.PyGraph, got {type(graph).__name__}"
+        )
 
     mapping = qaoa.cycle.wires_to_edges(graph)
```

Comment:
this function did not need to be modified?

Reply:
It doesn't need any modification as I already ensured that the results of all the subroutines are exactly the same either using RX or NX (to do so, I used some hacking like sorting the returned values...)